src/MongoDB.EntityFrameworkCore/Serializers/SerializationHelper.cs

```diff
@@ -106,74 +107,55 @@ internal static BsonSerializationInfo GetPropertySerializationInfo(IReadOnlyProp
         var serializer = CreateTypeSerializer(property.ClrType, property);
         if (property.IsPrimaryKey() && property.DeclaringEntityType.FindPrimaryKey()?.Properties.Count > 1)
         {
-            return BsonSerializationInfo.CreateWithPath(new[] {"_id", property.GetElementName()}, serializer, property.ClrType);
+            return BsonSerializationInfo.CreateWithPath(new[]
+            {
+                "_id", property.GetElementName()
+            }, serializer, property.ClrType);
         }
 
         return new BsonSerializationInfo(property.GetElementName(), serializer, property.ClrType);
     }
 
     private static IBsonSerializer CreateTypeSerializer(Type type, IReadOnlyProperty? property = null)
-    {
-        bool isNullable = type.IsNullableValueType();
-        if (isNullable)
+        => type switch
         {
-            type = Nullable.GetUnderlyingType(type);
```

Comment:
No need to handle nullables anymore?

Reply:
They are handled in the switch statement below in the CreateNullableSerializer pattern.